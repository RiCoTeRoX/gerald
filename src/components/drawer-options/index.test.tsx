import { DrawerContentComponentProps } from '@react-navigation/drawer'
import { render } from '@testing-library/react-native'
import React from 'react'

import DrawerOptions from '.'

describe('DrawerOptions', () => {
  it('renders correctly', () => {
    const mockedProps: DrawerContentComponentProps = {
      state: {
        index: 0,
        key: 'key',
        routeNames: ['routeNames'],
        routes: [{ key: 'key', name: 'name' }],
        stale: false,
        type: 'drawer',
        default: 'closed',
        history: [],
      },
      navigation: {
        closeDrawer: jest.fn(),
        dispatch: jest.fn(),
        goBack: jest.fn(),
        isFocused: jest.fn(),
        jumpTo: jest.fn(),
        navigate: jest.fn(),
        openDrawer: jest.fn(),
        reset: jest.fn(),
        toggleDrawer: jest.fn(),
        canGoBack: jest.fn(),
        getId: jest.fn(),
        getState: jest.fn(),
        getParent: jest.fn(),
        emit: jest.fn(),
        setParams: jest.fn(),
      },
      descriptors: {
        key: {
          route: {
            key: 'key',
            name: 'name',
            params: {},
          },

          navigation: {
            closeDrawer: jest.fn(),
            dispatch: jest.fn(),
            goBack: jest.fn(),
            isFocused: jest.fn(),
            jumpTo: jest.fn(),
            navigate: jest.fn(),
            openDrawer: jest.fn(),
            reset: jest.fn(),
            setOptions: jest.fn(),
            toggleDrawer: jest.fn(),
            canGoBack: jest.fn(),
            getId: jest.fn(),
            getState: jest.fn(),
            getParent: jest.fn(),
            setParams: jest.fn(),
            addListener: jest.fn(),
            removeListener: jest.fn(),
          },
          options: {
            drawerIcon: jest.fn(),
            drawerLabel: jest.fn(),
            drawerPosition: 'left',
            drawerType: 'back',
            header: jest.fn(),
            headerShown: false,
            title: '',
          },
          render: jest.fn(),
        },
      },
    }
    const { toJSON } = render(<DrawerOptions {...mockedProps} />)

    expect(toJSON()).toMatchSnapshot()
  })
})
