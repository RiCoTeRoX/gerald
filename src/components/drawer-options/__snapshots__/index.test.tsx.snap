// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerOptions renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#161629",
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        {
          "transform": [
            {
              "translateY": undefined,
            },
          ],
        },
      ]
    }
  >
    <View
      style={
        {
          "backgroundColor": "#161629",
          "borderTopLeftRadius": 20,
          "flex": 1,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;
